saleor/discount/utils.py

```diff
@@ -581,6 +582,19 @@ def get_current_products_for_rules(rules: "QuerySet[PromotionRule]"):
     return Product.objects.filter(Exists(variants.filter(product_id=OuterRef("id"))))
 
 
+def _create_new_rules(rules_to_add, variants_lock, rules_lock):
+    # base on what locks returned, filter out rules and variants that weren't locked
+    rules_to_add_batch = [
+        rv
+        for rv in rules_to_add
+        if rv.promotionrule_id in rules_lock and rv.productvariant_id in variants_lock
+    ]
+
+    return PromotionRule.variants.through.objects.bulk_create(
+        rules_to_add_batch, ignore_conflicts=True
```

Comment:
Do we need that ignore conflicts? There is only one constraint beside `pk` on this model, and I believe it's some old integration field with default set to `null`.